source/auth/auth.rst

```diff
@@ -1213,14 +1221,113 @@ mechanism
 mechanism_properties
     PROVIDER_NAME
         Drivers MUST allow the user to specify the name of the OIDC service to
-        use to obtain credentials. MUST be one of ["aws", "custom"].
+        use to obtain credentials. MUST be one of ["aws"].
+
+Interfaces
+``````````
+
+Custom Callback
+```````````````
+Drivers MUST provide a way for the callback to be either
+automatically canceled, or to cancel itself.  This can be as a timeout argument
+to the callback, a cancellation context passed to the callback, or some other
+language-appropriate mechanism.  The timeout deadline MUST be the same as the
+connection establishment deadline (TODO: Does that make sense?).
+
+Callbacks can be synchronous and/or asynchronous, depending on the driver
+and/or language.  Asynchronous callbacks should be preferred when other
+operations in the driver use asynchronous functions.
+
+The driver MUST pass the following information to the callback:
+
+- ``timeoutMS``: A timeout, in milliseconds, deadline, or ``timeoutContext``.
+- ``mechanismProperties``: All key-value pairs from the
+  AUTH_MECHANISM_PROPERTIES parameter.
+- ``invalidatedToken``: Optional. The previous token that was invalidated after
+  receiving a ``ReauthenticationRequired`` error.
+
+The callback MUST return an OIDC access token, formatted as a base64-encoded
+JWT.
+TODO: Is that correct?
```

Comment:
Not sure whether this TODO refers to this specifically, but “base64-encoded JWT” might be confusing language; JWTs consist of period-separated base64url-encoded parts already by definition, so “base64-encoded JWT” sounds like the expectation is that there is an additional round of base64 encoding happening on the full JWT.

```suggestion
The callback MUST return an OIDC access token in JWT format.
```

Reply:
(sorry, not sure what was up with Github here)